### PR TITLE
add gh action to validate the crd

### PR DESCRIPTION
**Pull request by friedrichwilken**
this adds a simple gh action that tries to apply the nats-crd to a k3d cluster to see if it is complient. very useful, if you work on the crd.

### CODE REVIEW

File: .github/workflows/validatecrd.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  validate:
+    runs-on: ubuntu-latest
```
Comment (mfaizanse) [inline .github/workflows/validatecrd.yml:11]: Do we need full ubuntu? is there any lite OS option like alpine etc.?

File: .github/workflows/validatecrd.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - uses: nolar/setup-k3d-k3s@v1
+        with:
+          version: v1.26.5
```
Comment (mfaizanse) [inline .github/workflows/validatecrd.yml:16]: this mean we have to upgrade k3s version manually?

Comment (friedrichwilken) [reply .github/workflows/validatecrd.yml:16]: that's the k8s version. I was wondering if this is confusing and if a comment could help. or to put it in other words: we are in control what version of k8s is used.

Comment (friedrichwilken) [reply .github/workflows/validatecrd.yml:17]: I added a comment to make the purpose of this version clear.

**Review by mfaizanse — APPROVED** [review]

**Pull request merged**